### ISSUE TITLE
How much data one need?

### ISSUE THREAD

**Issue by nikhilpandey360** [jocicmarko/ultrasound-nerve-segmentation#54]
I'm trying to detect objects using the U-Nets. I have around 100 images and each image contains  around 40-150 such objects. The image resolution is pretty good avg out to be around 3.5 MB. How do I calculate how much such images should I use so that I may avoid overfitting as well as underfitting?

**Comment by jocicmarko** [comment]
It really depends on many things, among them:

* what is your model complexity?
* do you regularize your model in some way?
* what is the input image size? (a resolution of an image which is 3.5 MB highly depends on image encoding: bmp, jpg, png...)
* how complex is the problem itself? what are the properties of the object you want to detect: their size, shape, are they distinguishable, etc? how many channels does the input image have?
* is it possible to do data augmentation in your problem?
* ...

Even if you knew all this, there is no formula that can tell you the number of images you need. Couple of things to try in the beginning:

* simple model, smaller images
* complex model, regularization, larger images
* smart data augmentation to create fake data

TLDR: if you find a definite answer to this question, you will be famous :)

**Comment by nikhilpandey360** [comment]
Got the answer. Thanks.
